R/render.R

```diff
@@ -264,3 +264,54 @@ infinite_moon_reader = function(moon, cast_from = '.') {
 #' @export
 #' @rdname inf_mr
 inf_mr = infinite_moon_reader
+
+
+#' Converting xaringan HTML slides to PDF
+#'
+#' This function uses the hosted docker image of the nodejs library 'decktape'
+#' to convert xaringan HTML slides (which are based on 'remark.js' under
+#' the hood) to PDF.
+#'
+#' @param xaringan_path The path to the xaringan HTML slides.
+#' @param pdf_path The desired output path of the PDF file.
+#' @param decktape_version Which `decktape` version should be used. Default is
+#'   to use the latest one.
+#' @param open_pdf Should the resulting PDF be opened with your system PDF
+#'   viewer?
+#'
+#' @section Docker:
+#' To run this function you need to have a working installation of
+#' [docker](https://www.docker.com/). For some operating systems you may need
+#' to [add yourself to the "docker" group](https://stackoverflow.com/questions/48957195/how-to-fix-docker-got-permission-denied-issue?noredirect=1&lq=1)
+#' and restart your machine.
+#'
+#' @section Decktape:
+#' This function uses the nodejs library [decktape](https://github.com/astefanutti/decktape).
+#' In fact the hosted docker version of decktape is used here.
+#' By default the latest decktape version is used.
+#' In case of errors you may want to try older decktape versions (version
+#' 2.8.0 works fine for example) using the `decktape_version` argument.
+#'
+#' @section Open-PDF:
+#' The opening of PDF relies on two points: Your operating system and your
+#' PDF viewer. Some setups work, some do not. It is recommended to use full paths
+#' as the internal function used (`animation:::auto_browse()`) does not like
+#' paths like "~/file.pdf".
+#'
+#' @export
+
+export_pdf = function(xaringan_path = NULL, pdf_path = NULL,
+                      decktape_version = NULL, open_pdf = FALSE) {
+
+  if (is.null(decktape_version)) {
+    system(sprintf("docker run --rm -t -v `pwd`:/slides -v $HOME:$HOME astefanutti/decktape %s %s",
+                   xaringan_path, pdf_path))
+  } else {
+    system(sprintf("docker run --rm -t -v `pwd`:/slides -v $HOME:$HOME astefanutti/decktape:%s %s %s",
+                   decktape_version, xaringan_path, pdf_path))
```

Comment:
> Are you able to merge the two system() calls into one to avoid the repetition of docker run --rm -t -v pwd:/slides -v $HOME:$HOME astefanutti/decktape?

Yes, did so. 

> BTW, what do `pwd`:/slides and $HOME:$HOME mean? Will they work on Windows?

AFAICT yes. The two parts you mentioned are required for some local writing permissions for `docker`. I do not know more details unfortunately. I modified the defaults of the `decktape` readme because they were not as robust as this notation.